server/routerlicious/packages/services/src/socketIoRedisPublisher.ts

```diff
@@ -24,12 +24,24 @@ export class SocketIoRedisTopic implements core.ITopic {
  * @internal
  */
 export class SocketIoRedisPublisher implements core.IPublisher {
-	private readonly redisClient: Redis.Redis;
+	private readonly redisClient: Redis.Redis | Redis.Cluster;
 	private readonly io: any;
 	private readonly events = new EventEmitter();
 
-	constructor(options: Redis.RedisOptions) {
-		this.redisClient = new Redis.default(options);
+	constructor(
+		options: Redis.RedisOptions,
+		enableClustering: boolean = false,
+		slotsRefreshTimeout: number = 50000,
+	) {
+		this.redisClient = enableClustering
```

Comment:
This method being exposed quite multiple time. We can create a static factory method in server shared/util location, and use it whenever you create this reidsClient. This will reduce duplicate code and make maintenance easier.

Reply:
Good idea, will make this change